### PR TITLE
FreeBSD compile fixes

### PR DESCRIPTION
**Pull request by raffenet**
A few compatibility fixes for build issues on FreeBSD.

### CODE REVIEW

**Comment by lanl-ompi** [comment]
bot:retest

File: include/freebsd/osd.h [hunk 1]
```diff
@@ -35,6 +35,7 @@
 
 #include <sys/endian.h>
 #include <pthread_np.h>
+#include <netinet/in.h>
```
Comment (shefty) [inline include/freebsd/osd.h]: Should this include be added to the .c file that actually needs it?

Comment (raffenet) [inline include/freebsd/osd.h]: Hmm, yes. That might be a more standard compliant fix given that the header should exist on all platforms.

Comment (raffenet) [inline include/freebsd/osd.h]: The comment here (https://github.com/ofiwg/libfabric/blob/master/util/pingpong.c#L345) indicates any protocol, which should be `0`, but then specifies `IPPROTO_TCP`. @bturrubiates is it necessary to do that?

Comment (raffenet) [inline include/freebsd/osd.h]: @piroux Actually my above comment applies to your original commit.

**Comment by raffenet** [comment]
FWIW, I updated the pull request to just modify the C file requiring the macros in `netinet/in.h`. There was more than just the instance of `IPPROTO_TCP` in there.

File: include/freebsd/osd.h (hunk 1 above)
Comment (bturrubiates) [inline include/freebsd/osd.h]: > The comment here (https://github.com/ofiwg/libfabric/blob/master/util/pingpong.c#L345) indicates any protocol, which should be 0, but then specifies IPPROTO_TCP. @bturrubiates is it necessary to do that?

I think the comment is just wrong. We want the out of band communication to be over TCP.

**Comment by shefty** [comment]
thanks

**Pull request merged**